### PR TITLE
Tried to implement xavier'10, weight initializaton

### PR DESCRIPTION
**Pull request by sadaf2605**
http://deeplearning.net/tutorial/mlp.html#weight-initialization  

I took f_in the number of inputs, from architecture command line 1 from "1:2:3:4:5", f_out the number of hidden units, 3 [as first and last one are input and output, so 5-2=3]

### CODE REVIEW

File: nntoolkit/create.py [hunk 1]
```diff
@@ -101,4 +109,4 @@ def main(nn_type, architecture, model_file):
 
 if __name__ == '__main__':
     args = get_parser().parse_args()
-    main(args.type, args.architecture, args.model_file)
+    main(args.type, args.architecture, args.model_file)
```
Comment (MartinThoma) [inline nntoolkit/create.py]: This newline should be here (see http://www.reddit.com/r/Python/comments/1zjugg/question_why_does_pep8_recommend_leaving_a_blank/)

File: nntoolkit/create.py [hunk 2]
```diff
@@ -48,18 +48,26 @@ def main(nn_type, architecture, model_file):
...
 
-        # TODO: the activation function could be here!
-        neurons = list(map(int, architecture.split(':')))
-
         for neurons_a, neurons_b in zip(neurons, neurons[1:]):
-            # TODO: right order?
-            W = [[random.random() for i in range(neurons_a)]
-                 for j in range(neurons_b)]
+
+            init_weight= 4.0*numpy.sqrt(6.0/(fan_in+fan_out))
```
Comment (MartinThoma) [inline nntoolkit/create.py]: Please use PEP8: spaces around '='! (I'll fix it this time)

File: nntoolkit/create.py [hunk 3]
```diff
@@ -86,11 +94,11 @@ def main(nn_type, architecture, model_file):
 
         model = {'type': 'mlp', 'layers': layers}
 
-    with open("model.yml", 'w') as f:
+    with open('model.yml', 'w') as f:
         yaml.dump(model, f, default_flow_style=False)
 
     # Create tar file
-    with tarfile.open("model.tar", "w:") as tar:
+    with tarfile.open(model_file, "w:") as tar:
```
Comment (MartinThoma) [inline nntoolkit/create.py]: Good! (I forgot it here ...)

File: nntoolkit/create.py [hunk 4]
```diff
@@ -48,18 +48,26 @@ def main(nn_type, architecture, model_file):
...
+    # TODO: the activation function could be here!
+    neurons = list(map(int, architecture.split(':')))
+
+    neurons_a, neurons_b=zip(neurons, neurons[1:])[0]
+ 
+    fan_in=neurons[0]
+    fan_out= len(neurons)-2  
+    model={}
+
+    if nn_type[0] == 'mlp':
```
Comment (MartinThoma) [inline nntoolkit/create.py]: Why did you change that from 'nn_type' to 'nn_type[0]'?

File: nntoolkit/create.py [hunk 5]
```diff
@@ -48,18 +48,26 @@ def main(nn_type, architecture, model_file):
 
     filenames = ["model.yml"]  # "input_semantics.csv", "output_semantics.csv"
 
-    if nn_type == 'mlp':
+    layers_binary = []
+    # TODO: the activation function could be here!
+    neurons = list(map(int, architecture.split(':')))
+
+    neurons_a, neurons_b=zip(neurons, neurons[1:])[0]
```
Comment (MartinThoma) [inline nntoolkit/create.py]: Why do you do that?

File: nntoolkit/create.py (hunk 4 above)
Comment (sadaf2605) [inline nntoolkit/create.py]: well i printed nn_type=='mlp' and it was false then I printed nn_type, it was a list, so I just did it :P

Comment (sadaf2605) [inline nntoolkit/create.py]: If that works without it, I don't think it is required!
